### ISSUE TITLE
Lazy loading is not triggered on filter input change in Dropdown

### ISSUE THREAD

**Issue by paveldayneko** [primefaces/primereact#2605]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case**

https://codesandbox.io/s/mystifying-bhaskara-v2twz?file=/src/demo/DropdownDemo.js

**Current behavior**
Filter input does not trigger virtual load

**Expected behavior**
onLazyLoad callback should be called on filter input change

**Minimal reproduction of the problem with instructions**

- Create virual scroll dropdown with lazy load
- Add filter
- Try to type some text in filter's input
- Nothing happens

* **PrimeReact version:**
7.1.0 and 7.2.0

* **Browser:** [Chrome]

 
* **Language:** [all]

**Comment by Catstein** [comment]
i'm having the same problem, i tried to force a solution by calling the "onFilter" method with the same "onlazyload" property call.. and it didn't work either :/

**Comment by melloware** [comment]
Yep right now it looks like the only thing filter is doing is setting the scroll index of lazy list back to 0.

```js
    const onFilterInputChange = (event) => {
        virtualScrollerRef.current && virtualScrollerRef.current.scrollToIndex(0);

        props.onFilterInputChange && props.onFilterInputChange(event);
    }
```

**Comment by vinirossa** [comment]
it's also happening with me, i've tried several things but i cant make it work :c

**Comment by dani-ras** [comment]
so are there any workarounds recommended?

**Comment by melloware** [comment]
I think it needs to be fixed in the component I just don't know what to fix it to.

**Comment by jonmferreira** [comment]
Hello budy, how are you?

I come to propose an alternative way of using dropdown with an asynchronous filter. Here's the proposal, I hope it helps you: https://react-qd8f2i-hurwba.stackblitz.io

**Comment by yabelozub** [comment]
Hi guys! I am facing the same problem, did kind like workaround with hardcoded non-clickable itemTemplate. Hope it will be fixed soon

**Comment by DieterMorar** [comment]
Hi, having the same issue. 
My workaround:
 
`onFilter={() => {
      const scrollerRef: HTMLElement | null = document.querySelector('.p-virtualscroller-content');
      if (scrollerRef?.style?.transform) {
        scrollerRef.style.transform = 'translate3d(0px, 0px, 0px)';
      }
    }}
`
